### ISSUE TITLE
Sharing Owner not Consistent/Correct

### ISSUE THREAD

**Issue by trichner** [fruux/sabre-dav#841]
Setup:
- user01 with one private calendar and one shared from user02 (owner)
- user02 with only one private calendar

Now when I check `d:owner` for the shared calendar with user01 logged in it is `	/principals/user01@example.com/`. Shouldn't this be `/principals/user02@example.com/`?

For comparison, if I do the same with Apple Calendar Server the `{DAV:}owner` property of the shared calendar points to user02.

Is `d:owner` equivalent to `{DAV:}owner`? If yes, shouldn't it point to the owner of the calendar?

**Comment by evert** [comment]
Hi @trichner , 

> Is d:owner equivalent to {DAV:}owner?

First, `{DAV:}owner` is how we in the sabre project specify a combination of an `owner` xml tag, in the `DAV:` xml namespace. So `d:owner` _might_ be the exact same thing, and it probably is, but it depends on the `xmlns:d` attribute appearing earlier in the document.

>  If yes, shouldn't it point to the owner of the calendar?

In the context of sharing, there is no specific meaning associated with `{DAV:}owner`. Owner does not necessarily imply 'this is the principal that is the original sharer of the resource'.

Because of this, different servers assign different meanings to `{DAV:}owner`. In the case of sabre/dav, with owner we want it to mean 'the person who owns _this_ instance of the resource' and not the original sharer.

If you want to find information about the original sharer of the resource, you should probably just fetch the `invite` property. This is assuming you are referring to this specification:

http://svn.calendarserver.org/repository/calendarserver/CalendarServer/trunk/doc/Extensions/caldav-sharing.txt

And not the newer one (that is not yet supported by every server):

https://tools.ietf.org/html/draft-pot-webdav-resource-sharing-03

Evert

**Comment by trichner** [comment]
Hi @evert ,
Thanks for the fast reply, I guess we'll check out the invite property.

While we're at it: I now implemented calendar sharing via the SabreDAV API but it still eludes me what for the `$sharee->href` property is.

Currently I'm setting it to the owner's mailto. But after staring at the RFC for a few minutes I'm not much wiser and could not figure out if this is the intended use, of if it's the current principal I should set.
```
$sharee->href = "mailto:$ownerPrincipalEmail"; // user02@example.com
$sharee->principal = $myPath; // principals/user01@example.com
...
$this->calendarBackend->updateInvites($calendarId, [$sharee]);
```

Thanks for a reply!
Thomas

**Comment by evert** [comment]
Hey Thomas,

It's a bit complicated because we didn't specify it exactly. Basically the `href` in share requests can be _any_ URI that uniquely identifies a user. So it could be a `mailto:`, a `http:` (relative principal or absolute), or even `ldap:`.

This is similar to the `ATTENDEE` property in iCalendar. It's basically specified as "any uri".

I suspect that most real calendaring clients will always just use `mailto:` addresses, at least right now. That's also currently the case for existing clients using the apple calendar sharing specification.

So that's the href. So what about the `principal` property? The principal property is _not_ something set by the client and it's not part of the sharing specification. It's a sabre/dav specific property.

Basically what happens is, if a caldav sharing client sets a `mailto:`, sabre/dav will check all the local principals to find out if there is a principal with that mailto address. If we found one, we automatically populate the `principal` property.

So the difference between `href` and `principal` is that:

1. href is determined by the client, principal is set by sabre/dav
2. href can be any URI, principal is _always_ a local path to a principal OR `null`.

**Comment by trichner** [comment]
Thanks, that answers my questions completely.
